server/repository/src/mock/ledger.rs

```diff
@@ -0,0 +1,209 @@
+use chrono::{NaiveDate, NaiveDateTime};
+
+use crate::{
+    mock::mock_name_store_b, InvoiceLineRow, InvoiceLineType, InvoiceRow, InvoiceStatus,
+    InvoiceType, ItemRow, StockLineRow,
+};
+
+// Mock ledger data, across 2 items, varying stock lines, invoice types and statuses
+pub fn mock_ledger_data() -> (
+    Vec<ItemRow>,
+    Vec<StockLineRow>,
+    Vec<InvoiceRow>,
+    Vec<InvoiceLineRow>,
+) {
+    let ledger_test_item_a = ItemRow {
+        id: "ledger_test_item".to_string(),
+        ..Default::default()
+    };
+    let ledger_test_item_b = ItemRow {
+        id: "ledger_test_item_b".to_string(),
+        ..Default::default()
+    };
+    let stock_line_a = StockLineRow {
+        id: "ledger_stock_line_a".to_string(),
+        item_link_id: ledger_test_item_a.id.clone(),
+        store_id: "store_a".to_string(),
+        ..Default::default()
+    };
+    let stock_line_b = StockLineRow {
+        id: "ledger_stock_line_b".to_string(),
+        item_link_id: ledger_test_item_a.id.clone(),
+        store_id: "store_a".to_string(),
+        ..Default::default()
+    };
+    let stock_line_a_item_b = StockLineRow {
+        id: "ledger_stock_line_a_item_b".to_string(),
+        item_link_id: ledger_test_item_b.id.clone(),
+        store_id: "store_a".to_string(),
+        ..Default::default()
+    };
+
+    let default_invoice = InvoiceRow {
+        store_id: "store_a".to_string(),
+        name_link_id: mock_name_store_b().id,
+        created_datetime: get_test_ledger_datetime(1),
+        ..Default::default()
+    };
+    let default_invoice_line = InvoiceLineRow {
+        item_link_id: "ledger_test_item".to_string(),
+        stock_line_id: Some(stock_line_a.id.clone()),
+        number_of_packs: 10.0,
+        pack_size: 1.0,
```

Comment:
can be good to use something other than 1 so we know we're calculating units in the views correctly. 10 is easy cause just add 0 everywhere in test numbers basically